src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -0,0 +1,417 @@
+                _thalorienGUID = thalorien->GetGUID();
+            me->SummonCreatureGroup(SUMMON_SUNWELL_DEFENDER);
+
+            _events.ScheduleEvent(EVENT_INTRO_0, 3s);
+        }
+    }
+
+    void Reset() override
+    {
+        me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+        _summons.DespawnAll();
+        _playerGUID.Clear();
+        _morlenGUID.Clear();
+        _thalorienGUID.Clear();
+        _events.Reset();
+    }
+
+    void ThalorienTalk(uint32 id)
+    {
+        if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+            thalorien->AI()->Talk(id);
+    }
+
+    void MorlenTalk(uint32 id)
+    {
+        if (Creature* morlen = ObjectAccessor::GetCreature(*me, _morlenGUID))
+            morlen->AI()->Talk(id);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_INTRO_0:
+                    ThalorienTalk(SAY_THALORIEN_1);
+
+                    _events.ScheduleEvent(EVENT_INTRO_1, 5s);
+                    break;
+                case EVENT_INTRO_1:
+                    ThalorienTalk(SAY_THALORIEN_2);
+
+                    _events.ScheduleEvent(EVENT_INTRO_2, 6s);
+                    break;
+                case EVENT_INTRO_2:
+                    ThalorienTalk(SAY_THALORIEN_3);
+
+                    _events.ScheduleEvent(EVENT_INTRO_3, 7s);
+                    break;
+
+                case EVENT_INTRO_3:
+                    ThalorienTalk(SAY_THALORIEN_4);
+
+                    _events.ScheduleEvent(EVENT_SALUTE, 4s);
+                    break;
+                case EVENT_SALUTE:
+                    for (auto& summon : _summons)
+                        if (Creature* creature = ObjectAccessor::GetCreature(*me, summon))
+                            creature->HandleEmoteCommand(EMOTE_ONESHOT_SALUTE);
+
+                    _events.ScheduleEvent(EVENT_DEFENDERS_RUN, 3s);
+                    break;
+                case EVENT_DEFENDERS_RUN:
+                {
+                    _events.ScheduleEvent(EVENT_DEFENDERS_RUN_2, 3s);
+                    _events.ScheduleEvent(EVENT_THALORIEN_GO, 2s);
+
+                    uint8 defendersCount = 0;
+                    for (auto& summon : _summons)
+                    {
+                        Creature* creature = ObjectAccessor::GetCreature(*me, summon);
+                        if (!creature)
+                            continue;
+
+                        if (creature->GetEntry() != NPC_SUNWELL_DEFENDER)
+                            continue;
+
+                        defendersCount++;
+                        if (defendersCount <= 5)
+                            creature->SetFacingTo(defenderReverse);
+                        else
```

Comment:
please add a "defendersCount < whateverMaxIndexValue" to ensure we don't access out of bounds defenderRun[ ] array, same in the rest of the script where defenderRun[ ] is accessed